src/library_webgl.js

```diff
@@ -3599,6 +3599,59 @@ var LibraryGL = {
     GLctx.sampleCoverage(value, !!invert);
   },
 
+  glMultiDrawArraysWEBGL: function(mode, firsts, firstsOffset, counts, countsOffset, drawcount) {
+    var firstsList = HEAP32.subarray(firsts >> 2, (firsts >> 2) + drawcount);
+    var countsList = HEAP32.subarray(counts >> 2, (counts >> 2) + drawcount);
+    GLctx.multiDrawWebgl['multiDrawArraysWEBGL'](
+      mode,
+      firstsList,
+      firstsOffset,
+      countsList,
+      countsOffset,
+      drawcount);
+  },
+  glMultiDrawArraysInstancedWEBGL: function(mode, firsts, firstsOffset, counts, countsOffset, instanceCounts, instanceCountsOffset, drawcount) {
+    var firstsList = HEAP32.subarray(firsts >> 2, (firsts >> 2) + drawcount);
+    var countsList = HEAP32.subarray(counts >> 2, (counts >> 2) + drawcount);
+    var instanceCountsList = HEAP32.subarray(instanceCounts >> 2, (instanceCounts >> 2) + drawcount);
+    GLctx.multiDrawWebgl['multiDrawArraysInstancedWEBGL'](
+      mode,
+      firstsList,
+      firstsOffset,
+      countsList,
+      countsOffset,
+      instanceCountsList,
+      instanceCountsOffset,
+      drawcount);
+  },
+  glMultiDrawElementsWEBGL: function(mode, counts, countsOffset, type, offsets, offsetsOffset, drawcount) {
+    var countsList = HEAP32.subarray(counts >> 2, (counts >> 2) + drawcount);
+    var offsetsList = HEAP32.subarray(offsets >> 2, (offsets >> 2) + drawcount);
+    GLctx.multiDrawWebgl['multiDrawElementsWEBGL'](
+      mode,
+      countsList,
+      countsOffset,
+      type,
+      offsetsList,
+      offsetsOffset,
+      drawcount);
+  },
+  glMultiDrawElementsInstancedWEBGL: function(mode, counts, countsOffset, type, offsets, offsetsOffset, instanceCounts, instanceCountsOffset, drawcount) {
+    var countsList = HEAP32.subarray(counts >> 2, (counts >> 2) + drawcount);
+    var offsetsList = HEAP32.subarray(offsets >> 2, (offsets >> 2) + drawcount);
+    var instanceCountsList = HEAP32.subarray(instanceCounts >> 2, (instanceCounts >> 2) + drawcount);
```

Comment:
All these `HEAP32.subarray` calls make me propose that the `WEBGL_multi_draw` extension be changed to a function signature form that takes an array+offset+length triplet, to avoid these costly subarrays. It would be strongly preferred that hot paths in WebGL are able to operate without generating temporary garbage.

Reply:
Looking back at the extensions, I just realized that they've already been phrased this way, and that these bindings aren't taking advantage of them. See other comments in this review.